packages/contracts/scripts/prebuild/export-smtchecker-contracts.ts

```diff
@@ -0,0 +1,46 @@
+import fs from "fs";
+import path from "path";
+
+const foundry_file = path.join(__dirname, "../../foundry.toml");
```

Comment:
Relative imports are brittle. I don't have a better suggestion for now other than recursively looping up until the `foundry.toml` file is found.

Reply:
I followed what you had [here](https://github.com/ubiquity/ubiquity-dollar/blob/372f0eb86e467af745d8ed5a1c9b615813c43a8b/packages/contracts/scripts/deploy/export.ts#L5) and some other places. Please confirm if you want me to change it.

Reply:
Ideally, yes. The further away the relative import is, the more shaky it is.

Reply:
What should be the limit of the search? If in some cases, there is no `foundry.toml`, we should prevent it from iterating until root, right?

Reply:
I'm not entirely sure. @rndquu perhaps you have an opinion here. I would default to something like
```javascript
const npmRoot = process.execSync('npm root').toString().trim();
const projectRoot = fs.resolve(npmRoot, "..");
const getProjectRoot = () => projectRoot;
```

Reply:
ChatGPT:

```typescript
import * as path from 'path';
import * as fs from 'fs';

function searchFoundryToml(): string | null {
  /**
   * Recursively searches for a file called "foundry.toml" in parent directories up to the root of the project.
   * Returns the absolute path of the file if found, or null if not found.
   */
  const rootDir = path.parse(process.cwd()).root;
  let currentDir = process.cwd();
  while (true) {
    const foundryTomlPath = path.join(currentDir, 'foundry.toml');
    if (fs.existsSync(foundryTomlPath) && fs.statSync(foundryTomlPath).isFile()) {
      return foundryTomlPath;
    }
    // Reached the root directory of the project, break the loop
    if (currentDir === rootDir) {
      break;
    }
    currentDir = path.dirname(currentDir);
  }
  return null;
}
```

Reply:
I would keep it simple and use a relative import

If the path fails to resolve in the future then CI action for smt (check my last comment) will fail and we will know that smth is wrong

Reply:
Okay sounds good.